graphs/breadth_first_search/is-graph-bipartite.cpp

```diff
@@ -0,0 +1,28 @@
+// Check if a graph is bipartite or not
+//BFS Solution
+
+class Solution {
+public:
+    bool isBipartite(vector<vector<int>>& graph) {
+        unordered_map<int, int> color;
+        for (int node = 0; node < graph.size(); ++node) {
+            if (color.count(node)) {
+                continue;
+            }
+            vector<int> stack{node};
+            color[node] = 0;
+            while (!stack.empty()) {
+                int curr = stack.back(); stack.pop_back();
+                for (const auto& neighbor : graph[curr]) {
+                    if (!color.count(neighbor)) {
+                        stack.emplace_back(neighbor);
+                        color[neighbor] = color[curr] ^ 1;
+                    } else if (color[neighbor] == color[curr]) {
+                        return false;
+                    }
+                }
+            }
+        }
+        return true;
+    }
+};
```

Comment:
Please add the main method so that the code can be runned and we can test it

Reply:
Hi, Sir I can do the changes what you want and will also add the proper comments wherever required in the code , along with good indentation to make it readable. Could you please assign this to me under hacktoberfest2022. 

Thank you 
Ritik Kumar Gupta